fhir2dataset/api_caller.py

```diff
@@ -60,32 +61,34 @@ def get_response(self, url: str):
         Arguments:
             url {str} -- url of the request
         """
-        # TODO : optimize count and paging
-        # count = self._get_count(url)
-        # if count == 0:
-        #     logger.warning(f"there is 0 matching resources for {url}")
-        # if url[-1] == "?":
-        #     url = f"{url}_count={count}"
-        # else:
-        #     url = f"{url}&_count={count}"
-        response = self._get_bundle_response(url)
-        self.status_code = response.status_code
-        try:
-            self.results = response.json()["entry"]
-        except KeyError as e:
-            logger.info(f"Got a KeyError - There's no {e} key in the json data we received.")
-            logger.debug(f"status code of KeyError response:\n{response.status_code}")
-            logger.debug(f"content of the KeyError response:\n{response.content}")
-        try:
-            self.next_url = None
-            for relation in response.json()["link"]:
-                if relation["relation"] == "next":
-                    self.next_url = relation["url"]
-                    break
-        except KeyError as e:
-            logger.info(f"Got a KeyError - There's no {e} key in the json data we received.")
-            logger.debug(f"status code of KeyError response:\n{response.status_code}")
-            logger.debug(f"content of the KeyError response:\n{response.content}")
+        self.next_url = None
+        if self.total is None:
```

Comment:
I wouldn't make `total` an attribute as you only use it in this method.
Maybe something like
```
if self._get_count(url) == 0:
    logger.info(f"there is {self.total} matching resources for {url}")
    return None (?)
<rest of the method>
```

Reply:
As discussed, self.total is used every time the function is called ie for every page linked to one url, it seems thus better to keep it an attribute